src/main/kotlin/org/jitsi/nlj/transform/node/outgoing/RetransmissionSender.kt

```diff
@@ -48,39 +48,47 @@ class RetransmissionSender : TransformerNode("Retransmission sender") {
     private val rtxStreamSeqNums: MutableMap<Long, Int> = mutableMapOf()
 
     private var numRetransmissionsRequested = 0
-    private var numRetransmittedPackets = 0
+    private var numRetransmittedRtxPackets = 0
+    private var numRetransmittedPlainPackets = 0
 
     override fun transform(packetInfo: PacketInfo): PacketInfo? {
         val rtpPacket = packetInfo.packetAs<RtpPacket>()
-        logger.cdebug { "${hashCode()} retransmitting packet with original ssrc " +
-                "${rtpPacket.ssrc}, original sequence number ${rtpPacket.sequenceNumber} and original " +
-                "payload type: ${rtpPacket.payloadType}" }
         numRetransmissionsRequested++
-        val rtxSsrc = associatedSsrcs[rtpPacket.ssrc] ?: run {
-            logger.cerror { "${hashCode()} could not find an associated RTX ssrc for original packet ssrc " +
-                    rtpPacket.ssrc }
-            packetDiscarded(packetInfo)
-            return null
-        }
-        val rtxPt = associatedPayloadTypes[rtpPacket.payloadType.toPositiveInt()] ?: run {
-            logger.cerror { "${hashCode()} could not find an associated RTX payload type for original payload type " +
-                    rtpPacket.payloadType }
-            packetDiscarded(packetInfo)
-            return null
+        var rtxRetransmission = false
+        val rtxPt = associatedPayloadTypes[rtpPacket.payloadType.toPositiveInt()]
```

Comment:
i think we can make this method cleaner if we split the logic for retransmitting with rtx and plain to some helper methods, then we get something like:

```
val rtxPt = associatedPayloadTypes[rtpPacket.payloadType.toPositiveInt()] ?: return retransmitPlain(packetInfo)
val rtxSsrc = associatedSsrcs[rtpPacket.ssrc] ?: return retransmitPlain(packetInfo)

return retransmitOverRtx(packetInfo, rtxPt, rtxSsrc)
```

Reply:
i tried this out some more, we'd end up with:
```
override fun transform(packetInfo: PacketInfo): PacketInfo? {
        val rtpPacket = packetInfo.packetAs<RtpPacket>()
        numRetransmissionsRequested++
        // note(george) this instance gets notified about both remote/local ssrcs (see Transeiver.addSsrcAssociation)
        // so, in the case of firefox, we end up having rtx (associated) ssrcs but no rtx (associated) payload type.
        val rtxPt = associatedPayloadTypes[rtpPacket.payloadType.toPositiveInt()] ?: return retransmitPlain(packetInfo)
        val rtxSsrc = associatedSsrcs[rtpPacket.ssrc] ?: return retransmitPlain(packetInfo)

        return retransmitRtx(packetInfo, rtxPt, rtxSsrc)
    }

    private fun retransmitRtx(packetInfo: PacketInfo, rtxPt: Int, rtxSsrc: Long): PacketInfo {
        val rtxSeqNum = rtxStreamSeqNums.merge(rtxSsrc, 1, Integer::sum)!!
        val rtpPacket = packetInfo.packetAs<RtpPacket>()
        logger.cdebug {
            "${hashCode()} sending RTX packet with ssrc $rtxSsrc with pt $rtxPt and seqNum " +
                "$rtxSeqNum with original ssrc ${rtpPacket.ssrc}, original sequence number " +
                "${rtpPacket.sequenceNumber} and original payload type: ${rtpPacket.payloadType}"
        }
        RtxPacket.addOriginalSequenceNumber(rtpPacket)
        rtpPacket.ssrc = rtxSsrc
        rtpPacket.payloadType = rtxPt
        rtpPacket.sequenceNumber = rtxSeqNum

        packetInfo.resetPayloadVerification()
        numRetransmittedRtxPackets++

        return packetInfo
    }

    private fun retransmitPlain(packetInfo: PacketInfo): PacketInfo {
        val rtpPacket = packetInfo.packetAs<RtpPacket>()
        logger.cdebug { "${hashCode()} plain retransmission packet with original ssrc " +
            "${rtpPacket.ssrc}, original sequence number ${rtpPacket.sequenceNumber} and original " +
            "payload type: ${rtpPacket.payloadType}" }

        numRetransmittedPlainPackets++
        // No work needed
        return packetInfo
    }
```

which i think ends up being clearer.

Reply:
much better, thanks